lib/db/icv.js

```diff
@@ -53,9 +63,120 @@ const convert = (conn, database, icvAxiom, options, params = {}) => {
   }).then(httpBody);
 };
 
+const validate = (conn, database, constraints, options, params = {}) => {
+  const headers = conn.headers();
+  headers.append('content-type', options.contentType || 'text/plain');
+  headers.append('accept', 'text/boolean');
+
+  const queryParams = {};
+  if (params.graphUri) {
+    queryParams['graph-uri'] = params.graphUri;
+  }
+  const query = qs.stringify(queryParams);
+  const uri = `${conn.uri(database, 'icv', 'validate')}${query.length > 0
+    ? `?${query}`
+    : ''}`;
+
+  return fetch(uri, {
+    method: 'POST',
+    body: constraints,
+    headers,
+  }).then(httpBody);
+};
+
+const validateInTx = (
+  conn,
+  database,
+  transactionId,
+  constraints,
+  options,
+  params = {}
+) => {
+  const headers = conn.headers();
+  headers.append('content-type', options.contentType || 'text/plain');
+  headers.append('accept', 'text/boolean');
+
+  const queryParams = {};
+  if (params.graphUri) {
+    queryParams['graph-uri'] = params.graphUri;
+  }
+  const query = qs.stringify(queryParams);
+  const uri = `${conn.uri(
+    database,
+    'icv',
+    transactionId,
+    'validate'
+  )}${query.length > 0 ? `?${query}` : ''}`;
+
+  console.info(uri);
+
+  return fetch(uri, {
+    method: 'POST',
+    body: constraints,
+    headers,
+  }).then(httpBody);
+};
+
+const violations = (conn, database, constraints, options, params = {}) => {
+  const headers = conn.headers();
+  headers.append('content-type', options.contentType || 'text/plain');
+  headers.append('accept', 'application/ld+json');
+
+  const queryParams = {};
+  if (params.graphUri) {
+    queryParams['graph-uri'] = params.graphUri;
+  }
+  const query = qs.stringify(queryParams);
+  const uri = `${conn.uri(database, 'icv', 'violations')}${query.length > 0
+    ? `?${query}`
+    : ''}`;
+
+  return fetch(uri, {
+    method: 'POST',
+    body: constraints,
+    headers,
+  }).then(httpBody);
+};
+
+const violationsInTx = (
```

Comment:
I'd argue, not strongly mind you, that this should be under the transaction namespace and not under the icv namespace.